File: agent/dotnet/src/Perper.Protocol/Cache/Notifications/CallTriggerNotification.cs
#pragma warning disable 8618, 0649
namespace Perper.Protocol.Cache.Notifications
{
    public class CallTriggerNotification : Notification
    {
        // NOTE: While ignite is case-insensitive with fields, it still duplicates the schema entries, hence the whole public/private dance
        private string call;
        private string @delegate;

        public string Call => call;
        public string Delegate => @delegate;

        public override string ToString()
        {
            return $"CallTriggerNotification({Call}, {Delegate})";
        }
    }
}

File: agent/python/perper/notifications/call_trigger_notification.py
from collections import OrderedDict
from pyignite import GenericObjectMeta
from pyignite.datatypes import String


class CallTriggerNotification(
    metaclass=GenericObjectMeta,
    schema=OrderedDict(
        [
            ("call", String),
            ("delegate", String),
        ]
    ),
):
    pass
